src/index.js

```diff
@@ -63,14 +72,12 @@ program.action(function(cmd, cmd2) {
       "instead"
     );
   } else {
-    var suggestion = didYouMean(cmd, commandNames);
-    suggestion = suggestion || didYouMean([cmd, cmd2].join(":"), commandNames);
-    if (suggestion) {
-      logger.error();
-      logger.error("Did you mean", clc.bold(suggestion) + "?");
+    if (!didYouMean(cmd, commandNames)) {
```

Comment:
why not:

```javascript
} else {
  if (!suggestCommands(cmd, commandNames)) {
    suggestCommands([cmd, cmd2].join(":"), commandNames);
  }
}

process.exit(1);
```

and in `suggestCommands` - return the `suggestion`. that way, all the uses of `didNotMean` are located in the same spot, (`suggestCommands`).

Maybe this is actually a terrible idea and breaks something I'm not aware of. But I'm struggling to figure out what the `!didYouMean` here really means - we have a command we don't know, it's not a renamed command so it's anything else. If we _did_ find a command using `didYouMean` we'd want to print it (which we do later). It's just a little difficult to follow the flow as-is because the logic of "do we need to print something", "what is it that we need to print", and "let's print the thing" are broken up in this form. I hope that all made sense...

Reply:
```js
var suggestion = didYouMean( cmd, commandNames );
suggestion = suggestion || didYouMean( [cmd, cmd2].join(':'), commandNames )
```
It was this logic that was being refactored as :-
```js
if (!didYouMean( cmd, commandNames) )
```
I guess the way you suggested would make the code appear redundant :thinking:
> It seems that things work as expected earlier :thinking:

Reply:
@bkendall Anyways it doesn't break things as of now
Can these changes be thought about as some sort of improvisation to the existing version :thinking:

Reply:
I'm very happy to accept these changes, but if this feels like a half solution, or "some sort" of improvement, we shouldn't settle until we have the best solution we can achieve.

This has encapsulated `didYouMean` within `suggestCommands`, which I believe to be a good idea. Since `suggestCommands` is in this file, it's _very_ easy to find out exactly what it's doing, happen to see how `didYouMean` works, and validate that `suggestCommands` does exactly what is expected - it prints a suggested command and returns it (if there was a suggestion).

In this current state, I have to know that `didYouMean` is returning a suggestion, that setting `cmd` to a new value is the correct change for how `suggestCommands` is used later, that having `suggestCommands` outside the scope of the `else` is correct, and _then_, if I'm debugging something, I have to trace `cmd` through a couple different paths before `suggestCommands` is run.

What I proposed isn't redundant; it limits the surface area we need to expose to the reader so they understand what's happening. `suggestCommands` does exactly one thing - if there's a command to be suggested, it prints it and returns it. Then, in the else statement ("if the `cmd` has not been renamed"), it makes sense to say "if we can't suggest a command for `cmd`, suggest a command for `cmd:cmd2`.

Also, in the basic case (e.g. `cmd` is `"halp"`), this reduces a loop from the complexity. If we do `didYouMean` and _then_ `suggestCommands` again, we're doing two loops (inside `didYouMean` - I don't know the implementation, but safe to say there's a loop of the array somewhere) when this can easily be reduced to one.